src/fdc/DiskManipulator.cc

```diff
@@ -410,24 +423,31 @@ void DiskManipulator::savedsk(const DriveSettings& driveData,
 	}
 }
 
-void DiskManipulator::create(std::span<const TclObject> tokens)
+static std::pair<MSXBootSectorType, std::vector<unsigned>> parsePartitionSizes(auto tokens)
 {
-	size_t totalSectors = 0;
 	MSXBootSectorType bootType = MSXBootSectorType::DOS2;
 	std::vector<unsigned> sizes;
 
-	for (const auto& token_ : view::drop(tokens, 3)) {
+	for (const auto& token_ : tokens) {
 		if (auto t = parseBootSectorType(token_.getString())) {
 			bootType = *t;
 		} else if (size_t sectors = parseSectorSize(token_.getString());
 		           sectors <= std::numeric_limits<unsigned>::max()) {
 			sizes.push_back(narrow<unsigned>(sectors));
-			totalSectors += sectors;
 		} else {
 			throw CommandException("Partition size too large.");
 		}
 	}
-	if (sizes.empty()) {
+
+	return {bootType, sizes};
+}
+
+void DiskManipulator::create(std::span<const TclObject> tokens)
+{
+	auto [bootType, sizes] = parsePartitionSizes(view::drop(tokens, 3));
+
+	size_t totalSectors = ranges::accumulate(sizes, size_t(0));
```

Comment:
Very much a detail. No need to change anything.
`ranges::accumulate(<range>, 0)` could be replaced with `sum(<range>)`

The former is closer to `std::accumulate`. So might be more familiar to c++ programmers who don't know the openMSX utility functions. OTOH there is no standard range version of accumulate.

The meaning of the later should also be clear. And it might be more future compatible with C++23/C++26: instead of adding `std::ranges::accumulate()` it seems more likely they will instead add `std::ranges::fold()` and `std::ranges::sum()` (https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2021/p2214r1.html#stdaccumulate-rangesfold)

Reply:
Good, if you’re looking for a fold and not so familiar with C++, accumulate is a confusing name with a confusing default operator, glad they’re changing that.

However with `sum` I can not express compactly that I want to sum unsigned types using a `size_t` accumulator. Unless I add a projection function which converts the type… I guess it’s more explicit that way, but also more verbose.

I could also implement it with `ranges::reduce` btw, that would potentially even be more efficient and proper, but accumulate has a longer pedigree and its name implies summing (which we happen to do here) so I chose that one.

Reply:
I overlooked the widening.

In any case, no need to change anything, current code is fine.
`sum()` with a projection would also be fine, you decide.